src/core/utils/CustomMulticallJsonRpcProvider.ts

```diff
@@ -1,30 +1,31 @@
-import { TransactionRequest } from 'ethers'
+import { JsonRpcApiProviderOptions, TransactionRequest } from 'ethers'
 import { Chain } from '../constants/chains'
 import { CustomJsonRpcProvider } from './customJsonRpcProvider'
-import { logger } from './logger'
 import { MulticallQueue } from './multicall'
 
+export interface CustomTransactionRequest extends TransactionRequest {
+  blockTag?: number
```

Comment:
`blockTag?: BlockTag` is already part of `TransactionRequest`, and it already accepts `number` as a possible value.

This will only restrict it so that you can only pass a number, which can be annoying if you get a block number from a provider as a `bigint` or `string`.

Any reason why we need this interface?